File: java/acronym/src/main/java/Acronym.java
import java.util.StringTokenizer;

class Acronym {

    private final String acronym;

    Acronym(String phrase) {
        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(phrase, " _-");

        while (st.hasMoreTokens())
            sb.append(Character.toUpperCase(st.nextToken().charAt(0)));

        acronym = sb.toString();
    }

    String get() { return acronym; }
}

File: javascript/acronym/acronym.js
const acronym_letter_pattern =
  /(?<=^|[_ -])([a-z])/gi;

export const parse = (phrase) => {

  let acronym = "";
  let match;

  while (match = acronym_letter_pattern.exec(phrase))
    acronym += match[0].toUpperCase();

  return acronym;
};
